packages/components/src/Select/Select.tsx

```diff
@@ -1,30 +1,185 @@
-import React from 'react';
+import React, { useRef } from 'react';
+import { HiddenSelect, useSelect } from '@react-aria/select';
+import { useListBox, useOption } from '@react-aria/listbox';
+import { useOverlay, DismissButton } from '@react-aria/overlays';
+import { FocusScope } from '@react-aria/focus';
+import { mergeProps } from '@react-aria/utils';
+import { useFocus } from '@react-aria/interactions';
+import { SelectState, useSelectState } from '@react-stately/select';
+import { useButton } from '@react-aria/button';
+import { SelectProps } from '@react-types/select';
+
+import { PolymorphicComponent, PolymorphicProps } from '@marigold/types';
 import { ArrowDown } from '@marigold/icons';
 import { useStyles } from '@marigold/system';
-import { ComponentProps } from '@marigold/types';
-import { Box } from '../Box';
 
-export type SelectProps = { variant?: string } & ComponentProps<'select'>;
+import { Box, BoxOwnProps } from '../Box';
+import { Text } from '../Text';
+
+export type SelectOwnProps = {
+  variant?: string;
+  label?: string;
+  disabled?: boolean;
+} & BoxOwnProps &
+  SelectProps<any>;
+export type SelectPolymorphicProps = PolymorphicProps<SelectOwnProps, 'select'>;
```

Comment:
Why is this component polymorphic?

Reply:
First I have to learn more about Polymorphic 😬